### ISSUE TITLE
KafkaConsumer.subscribe(topics='t', pattern='^t') should throw ValueError not AssertionError

### ISSUE THREAD

**Issue by jeffwidman** [dpkp/kafka-python#891]
For `KafkaConsumer.subscribe()` either `topics` or `pattern` must be specified but not both. While experimenting, I noticed that when I break this contract, I get an AssertionError.

Since this is caused by bad input, and not bad code, this should throw ValueError not AssertionError. 

Assertions often get stripped out in production, removing the safety net of this check.

Happy to put together a PR if you want, but wanted to doublecheck you're on board with this change.

**Comment by dpkp** [comment]
makes sense and on board! I am a fan of asserts to document assumptions, but I think you're right that we're probably abusing them in cases like this.

**Comment by dpkp** [comment]
Also note that technically what is raised is `IllegalStateError` which isn't strictly an assertion. This was done to mirror the java client, which I believe raises a similar error in this case.

**Comment by dpkp** [comment]
#3042
